packages/nextjs/app/page.tsx

```diff
@@ -3,8 +3,44 @@
 import Link from "next/link";
 import type { NextPage } from "next";
 import { BugAntIcon, MagnifyingGlassIcon } from "@heroicons/react/24/outline";
+import { useScaffoldReadContract } from "~~/hooks/scaffold-eth";
 
 const Home: NextPage = () => {
+  const {
+    data: checkedInCounter,
+    error,
+    isLoading: contractLoading,
+  } = useScaffoldReadContract({
+    contractName: "BatchRegistry",
+    functionName: "checkedInCounter",
+  });
+
+  const renderCheckedInCounterSection = () => {
+    if (contractLoading) {
+      return (
+        <div className="animate-pulse">
+          <div className="h-6 bg-gray-200 rounded w-48 mb-4"></div>
+          <div className="h-4 bg-gray-200 rounded w-24"></div>
+        </div>
+      );
+    }
```

Comment:
I think this might be not needed, you could conditionally render in line 39 instead.